jormungandr/src/fragment/process.rs

```diff
@@ -41,82 +48,136 @@ impl Process {
+                                }
+                                TransactionMsg::GetStatuses(fragment_ids, reply_handle) => {
+                                    let mut statuses = HashMap::new();
+                                    pool.logs().logs_by_ids(fragment_ids).into_iter().for_each(
+                                        |(fragment_id, log)| {
+                                            statuses.insert(fragment_id, log.status().clone());
+                                        },
+                                    );
+                                    reply_handle.reply_ok(statuses);
+                                }
+                                TransactionMsg::SelectTransactions {
+                                    pool_idx,
+                                    ledger,
+                                    ledger_params,
+                                    selection_alg,
+                                    reply_handle,
+                                    soft_deadline_future,
+                                    hard_deadline_future,
+                                } => {
+                                    let contents = pool
+                                        .select(
+                                            pool_idx,
+                                            ledger,
+                                            ledger_params,
+                                            selection_alg,
+                                            soft_deadline_future,
+                                            hard_deadline_future,
+                                        )
+                                        .await;
+                                    reply_handle.reply_ok(contents);
+                                }
+                            }
+                        }
                     }
-                    TransactionMsg::SelectTransactions {
-                        pool_idx,
-                        ledger,
-                        ledger_params,
-                        selection_alg,
-                        reply_handle,
-                        soft_deadline_future,
-                        hard_deadline_future,
-                    } => {
-                        let contents = pool
-                            .select(
-                                pool_idx,
-                                ledger,
-                                ledger_params,
-                                selection_alg,
-                                soft_deadline_future,
-                                hard_deadline_future,
-                            )
-                            .await;
-                        reply_handle.reply_ok(contents);
+                    _ = &mut wakeup => {
+                        let dir = persistent_log_dir.as_ref().unwrap();
+                        let file = open_log_file(dir.as_ref())?;
+                        pool.set_persistent_log(file);
```

Comment:
I forgot: `wakeup` needs to be re-assigned with a fresh future, to the effect of:
```suggestion
                        pool.set_persistent_log(file);
                        wakeup = hourly_wakeup(persistent_log_dir.is_some());
```

Might not work exactly like that due to the on-stack pin, so just convert it to a boxed future if necessary.

Reply:
And maybe the function should return the current hour as a byproduct, to avoid any discrepancy in the file names, but this is finessing.